browser/src/control/jsdialog/Definitions.Menu.ts

```diff
@@ -524,6 +524,114 @@ function generateLayoutPopupGrid(unoCommand: string): GridWidgetJSON {
 	return grid as any as GridWidgetJSON;
 }
 
+function generatePictureBrightnessMenu(
+	unoCommand: string,
+): Array<MenuDefinition> {
+	const brightnessValues = [
+		{ value: -40, text: _('minusforty') },
+		{ value: -20, text: _('minustwenty') },
+		{ value: 0, text: _('zero') },
+		{ value: 20, text: _('twenty') },
+		{ value: 40, text: _('forty') },
+	];
+
+	const menuItems: Array<MenuDefinition> = [];
+
+	for (let i = 0; i < brightnessValues.length; i++) {
+		menuItems.push({
+			id: brightnessValues[i].text,
+			uno:
+				'.uno:' + unoCommand + '?Brightness:short=' + brightnessValues[i].value,
+			text: '' + brightnessValues[i].value,
+			img: 'insertgraphic',
+		} as MenuDefinition);
+	}
+
+	return menuItems;
+}
+
+function generatePictureContrastMenu(
+	unoCommand: string,
+): Array<MenuDefinition> {
+	const contrastValues = [
+		{ value: -40, text: _('minusforty') },
+		{ value: -20, text: _('minustwenty') },
+		{ value: 0, text: _('zero') },
+		{ value: 20, text: _('twenty') },
+		{ value: 40, text: _('forty') },
+	];
+
+	const menuItems: Array<MenuDefinition> = [];
+
+	for (let i = 0; i < contrastValues.length; i++) {
+		menuItems.push({
+			id: contrastValues[i].text,
+			uno: '.uno:' + unoCommand + '?Contrast:short=' + contrastValues[i].value,
+			text: '' + contrastValues[i].value,
+			img: 'insertgraphic',
+		} as MenuDefinition);
+	}
+
+	return menuItems;
+}
+
+function generatePictureTransparencyMenu(
+	unoCommand: string,
+): Array<MenuDefinition> {
+	const transparencyValues = [
+		{ value: 0, text: _('zero') },
+		{ value: 15, text: _('fifteen') },
+		{ value: 30, text: _('thirty') },
+		{ value: 50, text: _('fifty') },
+		{ value: 65, text: _('sixtyfive') },
+		{ value: 80, text: _('eighty') },
+		{ value: 95, text: _('ninetyfive') },
+	];
+
+	const menuItems: Array<MenuDefinition> = [];
+
+	for (let i = 0; i < transparencyValues.length; i++) {
+		menuItems.push({
+			id: transparencyValues[i].text,
+			uno:
+				'.uno:' +
+				unoCommand +
+				'?Transparency:short=' +
+				transparencyValues[i].value,
+			text: '' + transparencyValues[i].value,
+			img: 'insertgraphic',
+		} as MenuDefinition);
+	}
+
+	return menuItems;
+}
+
+function generatePictureColorModeMenu(
+	unoCommand: string,
+): Array<MenuDefinition> {
+	//see enum GraphicDrawMode in include/vcl/GraphicAttributes.hxx
+	const colorModeValues = [
```

Comment:
can we just create static JSON like in https://github.com/CollaboraOnline/online/pull/12381/files#diff-1149cd1caa3b13b680d847c2a0bf5020791b738a3740f4cf8baf166cb60c87e9R466

it is not big and will be shorter than the whole function :)